tracker/tracker-assist/src/Assist.ts

```diff
@@ -243,120 +293,133 @@ export default class Assist {
     if (this.options.config) {
       peerOptions['config'] = this.options.config
     }
+   
     const peer = this.peer = new Peer(peerID, peerOptions)
-    // app.debug.log('Peer created: ', peer)
+
     // @ts-ignore
     peer.on('error', e => app.debug.warn('Peer error: ', e.type, e))
     peer.on('disconnected', () => peer.reconnect())
-    peer.on('call', (call) => {
-      app.debug.log('Call: ', call)    
-      if (this.callingState !== CallingState.False) {
-        call.close()
-        //this.notifyCallEnd() // TODO: strictly connect calling peer with agent socket.id
-        app.debug.warn('Call closed instantly bacause line is busy. CallingState: ', this.callingState)
+
+    const handleCallEnd = (callId: string) => {
+      const call = this.calls[callId]
+      if (!call) {
+        app.debug.warn('Call end from unknown callId', callId)
         return
       }
+      app.debug.log('Handle Call End')
+      call.close()
+      callUI?.remove()
+      annot?.remove()
+      callUI = null
+      annot = null
+
+      this.emit('UPDATE_SESSION', { agentIds: [], isCallActive: false, })
+      this.setCallingState(CallingState.False, call)
+      this.options.onCallStart &&  this.options.onCallStart()
+      this.stopLocalStreams()
+    }
 
-      const setCallingState = (newState: CallingState) => {
-        if (newState === CallingState.True) {
-          sessionStorage.setItem(this.options.session_calling_peer_key, call.peer)
-        } else if (newState === CallingState.False) {
-          sessionStorage.removeItem(this.options.session_calling_peer_key)
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    peer.on('call', async (call) => {
+      const initiateCallEnd = () => {
+        this.notifyCallEnd()
+        handleCallEnd(call.peer)
+      }
+      app.debug.log('Call: ', call)
+      try {
+        if (!this.lStream[call.peer]) {
+          app.debug.log('starting new stream for', call.peer)
+          this.lStream[call.peer] = await RequestLocalStream()
+          this.calls[call.peer] = call
         }
-        this.callingState = newState
+      } catch (e) {
+        app.debug.warn('Audio mediadevice request error:', e)
+        initiateCallEnd()
+      }
+      this.onRemoteCallEnd = (): void => {
```

Comment:
It would probably make sense to make a one function that relies in its behavior on the calling state. 
That will require  also pulling `confirmCall ` to an upper context (this makes sense because it is common for every peer call).